From 1b87c0e53efdccec4c05d2b92699f49cd3d6ec79 Mon Sep 17 00:00:00 2001
From: Lans Zhang <jia.zhang@windriver.com>
Date: Wed, 16 Aug 2017 19:52:37 +0800
Subject: [PATCH] cert-to-efi-hash-list: support to build with openssl-1.1.x

X509 becomes opaque since openssl-1.1.x and thus uses the equivalent
function i2d_re_X509_tbs() instead to encode tbs portion of the certificate.

Signed-off-by: Lans Zhang <jia.zhang@windriver.com>
---
 cert-to-efi-hash-list.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cert-to-efi-hash-list.c b/cert-to-efi-hash-list.c
index 3792553..2f2584c 100644
--- a/cert-to-efi-hash-list.c
+++ b/cert-to-efi-hash-list.c
@@ -135,7 +135,11 @@ main(int argc, char *argv[])
         X509 *cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL);
 	unsigned char *cert_buf = NULL;
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	int cert_len = i2d_X509_CINF(cert->cert_info, &cert_buf);
+#else
+	int cert_len = i2d_re_X509_tbs(cert, &cert_buf);
+#endif
 	ERR_print_errors_fp(stdout);
 
 	int len, digest_len, time_offset;
-- 
2.7.5

